### PR TITLE
Removing analytics v1 and optimizing v2

### PR DESCRIPTION
**Pull request by bcdurak**
## Describe changes
This PR focuses on removing the first version of our analytics module and optimizing the second version. Some of the most critical changes include the following:

- `analytics-python` has been removed as a dependency.
- The `utils.analytics_utils.py` and all its exclusive helper functions have been completely removed.
- The `analytics` module (v2) is completely reworked.
  - As we used to set the `sync_mode` to `True` in our previous usage of v2, the client-consumer design was deemed useless and thus removed.
  - The remaining models, enums, and helper functions have been moved to the `analytics` module.
- There is a new context manager called `analytics_disabler` and as its name suggests, it sets the env variable `ZENML_ANALYTICS_OPT_IN` to `False` and disables the analytics for a given context. This is used when default stacks, components, and flavors are generated.
- All of the `track`, `group`, and `identify` calls have been adjusted accordingly. 
- The analytics events which were only used in v1 have been removed.
- A bugfix that makes it now possible to correctly use `zenml init` as a source in the `identify` event. 
- Updated the corresponding section in the docs.

## Pre-requisites
Please ensure you have done the following:
- [X] I have read the **CONTRIBUTING.md** document.
- [X] If my change requires a change to docs, I have updated the documentation accordingly.
- [x] If I have added an integration, I have updated the [integrations](https://docs.zenml.io/stacks-and-components/component-guide) table and the [corresponding website section](https://zenml.io/integrations).
- [x] I have added tests to cover my changes.

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Other (add details above)

### CODE REVIEW

**Review by strickvl — COMMENTED** [review]
Most of my questions are around the removal of tracking (though I welcome it, to be clear). Otherwise this looks good to me, assuming you've ensured that it's working fine on the Segment end?

File: src/zenml/analytics/utils.py [hunk 1]
```diff
@@ -0,0 +1,243 @@
...
+        identify(metadata={"email": email})
+
+    # Track that the user answered the prompt
+    track(
+        AnalyticsEvent.OPT_IN_OUT_EMAIL,
+        {"opted_in": opted_in, "source": source},
+    )
+
+
+class analytics_disabler:
```
[the remark this answers is not in the document]
Comment (bcdurak) [reply src/zenml/analytics/utils.py:98]: `mypy` actually does not complain. As we use it as a decorator, I would keep this as is.

[the remark this answers is not in the document]
Comment (strickvl) [reply src/zenml/analytics/utils.py:98]: Got it. I had to change mine on `mlstacks` for this reason but not a big deal so def leave it if you like it that way.

File: src/zenml/analytics/utils.py
[the remark this answers is not in the document]
Comment (bcdurak) [reply src/zenml/analytics/utils.py:1]: I wanted to separate the analytics module out of the main repository as much as possible for now, considering we might rip it out in the future.

**Review by strickvl — APPROVED** [review]

**Pull request merged**